src/fragmentarium/ui/search/NumberSearchForm.tsx

```diff
@@ -1,57 +1,33 @@
 import React, { Component } from 'react'
-import { stringify } from 'query-string'
-import { Form, Button, Row, Col } from 'react-bootstrap'
-import { withRouter, RouteComponentProps } from 'react-router-dom'
-import { History } from 'history'
+import { Col, Form, Row } from 'react-bootstrap'
 
 type Props = {
-  number: string | null | undefined
-  history: History
-} & RouteComponentProps
-
-class FragmentSearchForm extends Component<Props> {
-  state = {
-    number: this.props.number || '',
-  }
+  onChange(searchForm: string, searchQuery: string): void
+  getState(key: string): string
+}
 
+class NumberSearchForm extends Component<Props> {
   onChange = (event) => {
-    this.setState({
-      number: event.target.value,
-    })
-  }
-
-  submit = (event) => {
-    event.preventDefault()
-    this.props.history.push(
-      `/fragmentarium/search/?${stringify({
-        number: this.state.number,
-      })}`
-    )
+    this.props.onChange('number', event.target.value || '')
   }
 
   render() {
     return (
-      <Form onSubmit={this.submit}>
+      <Form>
         <Form.Group as={Row} controlId="number">
-          <Col sm={2} />
-          <Col sm={7}>
+          <Col sm={{ span: 10, offset: 2 }}>
             <Form.Control
               type="text"
-              value={this.state.number}
+              value={this.props.getState('number')}
```

Comment:
This will most likely mess up updating the component. Why not just pass the number as a prop?